CMakeLists.txt

```diff
@@ -315,6 +310,31 @@ option(USE_OPENSSL "Enable use of OpenSSL if available" ON)
 option(SEND_TELEMETRY_DEFAULT "The default value for whether to send telemetry" ON)
 option(REGRESS_CHECKS "PostgreSQL regress checks through installcheck" ON)
 
+# Option to enable assertions. Note that if we include headers from a
+# PostgreSQL build that has assertions enabled, we might inherit that
+# setting without explicitly enabling assertions via the ASSERTIONS
+# option defined here. Thus, this option is mostly useful to enable
+# assertions when the PostgreSQL we compile against has it disabled.
+option(ASSERTIONS "Compile with assertion checks (default OFF)" OFF)
+
+file(READ ${PG_INCLUDEDIR}/pg_config.h PG_CONFIG_H)
+string(REGEX MATCH "#define USE_ASSERT_CHECKING 1" PG_USE_ASSERT_CHECKING ${PG_CONFIG_H})
```

Comment:
Even though unlikely that the file does not exist, you should probably have a check for the file before reading it and bail out if the file could not be found. Other options provided to CMake could change the include path to make sure that `pg_config.h` can be found but it is not in the expected location.

A more safe approach might be to use `find_path` and provide the `INCLUDE_PATHS` property to the `HINTS` parameter, if you wish. This should be guaranteed to work as long as the code compiles.

I think a check for the file is needed though, to avoid the not-entirely-obvious error message.

Reply:
I guess I can add that check, but it seems a bit unnecessary IMO. We should be able to trust `PG_INCLUDEDIR` at this point or something else is severely broken, which is a different problem in that case. The CMake run will fail if the file is not found, but this is very unlikely.

At this point we already did a `find_program` to figure out the location of `pg_config`, which in turn provides us with the `PG_INCLUDEDIR` path. If there's no `pg_config.h` in that location then the PostgreSQL installation is broken, and I don't think it is worth writing code for that case.

Reply:
Since I find the situation unlikely, I do not think it is very important to add the check. Just a suggestion.